File: Data Structures & Algorithms/is-anagram/submission-0.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length()) {
            return false;
        }

        char[] sArr = s.toCharArray();
        char[] tArr = t.toCharArray();

        int[] freqs = new int[26];

        for (int i = 0; i < sArr.length; i++) {
            freqs[sArr[i] - 'a']++;
            freqs[tArr[i] - 'a']--;
        }

        for (int freq: freqs) {
            if (freq != 0) {
                return false;
            }
        }
        return true;
    }
}

File: Data Structures & Algorithms/is-anagram/submission-0.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if not s or not t or len(s) != len(t):
            return False

        freqs = [0] * 26
        for i in range(len(s)):
            freqs[ord(s[i]) - ord('a')] += 1
            freqs[ord(t[i]) - ord('a')] -= 1
        
        return all(freq==0 for freq in freqs)